Userland/Libraries/LibGfx/ICCProfile.cpp

```diff
@@ -1019,13 +1019,213 @@ ErrorOr<void> Profile::read_tag_table(ReadonlyBytes bytes)
     return {};
 }
 
+static bool is_xCLR(ColorSpace color_space)
+{
+    switch (color_space) {
+    case ColorSpace::TwoColor:
+    case ColorSpace::ThreeColor:
+    case ColorSpace::FourColor:
+    case ColorSpace::FiveColor:
+    case ColorSpace::SixColor:
+    case ColorSpace::SevenColor:
+    case ColorSpace::EightColor:
+    case ColorSpace::NineColor:
+    case ColorSpace::TenColor:
+    case ColorSpace::ElevenColor:
+    case ColorSpace::TwelveColor:
+    case ColorSpace::ThirteenColor:
+    case ColorSpace::FourteenColor:
+    case ColorSpace::FifteenColor:
+        return true;
+    default:
+        return false;
+    }
+}
+
+ErrorOr<void> Profile::check_required_tags()
+{
+    // ICC v4, 8 Required tags
+
+    // ICC v4, 8.2 Common requirements
+    // "With the exception of DeviceLink profiles, all profiles shall contain the following tags:
+    //  - profileDescriptionTag (see 9.2.41);
+    //  - copyrightTag (see 9.2.21);
+    //  - mediaWhitePointTag (see 9.2.34);
+    //  - chromaticAdaptationTag, when the measurement data used to calculate the profile was specified for an
+    //    adopted white with a chromaticity different from that of the PCS adopted white (see 9.2.15).
+    //  NOTE A DeviceLink profile is not required to have either a mediaWhitePointTag or a chromaticAdaptationTag."
+    // profileDescriptionTag, copyrightTag are required for DeviceLink too (see ICC v4, 8.6 DeviceLink profile).
+    // profileDescriptionTag, copyrightTag, mediaWhitePointTag are required in ICC v2 as well.
+    // chromaticAdaptationTag isn't required in v2 profiles as far as I can tell.
+
+    if (!m_tag_table.contains(profileDescriptionTag))
+        return Error::from_string_literal("ICC::Profile: required profileDescriptionTag is missing");
+
+    if (!m_tag_table.contains(copyrightTag))
+        return Error::from_string_literal("ICC::Profile: required copyrightTag is missing");
+
+    if (device_class() != DeviceClass::DeviceLink) {
+        if (!m_tag_table.contains(mediaWhitePointTag))
+            return Error::from_string_literal("ICC::Profile: required mediaWhitePointTag is missing");
+
+        // FIXME: Check for chromaticAdaptationTag after figuring out when exactly it needs to be present.
+    }
+
+    auto has_all_tags = [&]<class T>(T tags) { return all_of(tags, [&](auto& key) { return m_tag_table.contains(key); }); };
```

Comment:
Two returns on one line is close to unreadable, add some line breaks, they're free :)

Reply:
Done.